packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -56,6 +61,19 @@ export class LazyLoader {
   }
 }
 
+/**
+ * Facades variant mapping data.
+ */
+export class VariantData {
+  name: string;
+  index: number;
+  materialVariants = new Array<number>();
+  constructor(name: string, index: number) {
```

Comment:
I don't think you need this in TS.

Reply:
I wanted to ensure that neither of these inputs gets missed so TS is enforcing that they are assigned on construction.

Reply:
I believe that's already enforced; you would need to use `name?: string;` to allow `name` to be  undefined.

Reply:
letting it be undefined is what I wanted to avoid and it if that's not allowed it needs to be defined on construction. I converted it to an interface, so now it can be defined with {} braces but I don't recommend this as those to values should always inserted at the same time.